snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/SnomedDeletionPlan.java

```diff
@@ -84,11 +82,8 @@ private void internalMarkForDeletion(final Collection<? extends CDOObject> items
 		deletedItems.addAll(items);
 	}
 	
-	public Collection<Description> getDirtyDescriptions() {
-		return Collections.unmodifiableCollection(descriptionsToUpdate);
-	}
-	
-	public void addDirtyDescription(Description... descriptions) {
-		descriptionsToUpdate.addAll(Sets.newHashSet(descriptions));
+	@Override
+	public String toString() {
+		return rejectionReasons.size() < 2 ? Iterables.getFirst(rejectionReasons, "") : Joiner.on(", ").join(rejectionReasons);
```

Comment:
`Joiner` will handle an empty or single-element `Iterable` in the expected manner:

```java
if (parts.hasNext()) {
  appendable.append(toString(parts.next()));
  while (parts.hasNext()) {
      appendable.append(separator);
      appendable.append(toString(parts.next()));
  }
}
```